### PR TITLE
Give python-docker-boefjes the possibility to use modules that are not part of OpenKAT

### LINKED ISSUE #3582: New docker-boefjes that use modules not used by OpenKAT throw exceptions which makes them not get included in the KAT-alogus.
**Describe the bug**
When a new docker-boefje gets developed, modules of that boefje can get specified inside that boefje's Docker file. But when a module gets added that is not already part of OpenKAT's default python modules, KAT-alogus will throw an error and not include the boefje in the KAT-alogus. 

**To Reproduce**

1. Create a new docker-boefje that uses a module not already used by OpenKAT (for example `ssdpy`)
2. run a clean `make kat` to see everything build successfully. 
3. When you go to the KAT-alogus tab, you will see that your new boefje is not available in the list, and when you look inside the KAT-alogus' logs, errors pop up.

**Expected behavior**
The boefje should be included in the KAT-alogus because OpenKAT should not require the boefje's dependencies since the boefje will have them installed inside its own docker container.

**Screenshots**
KAT-alogus' errors:
image

**OpenKAT version**
Note the release tag (and if possible: the installation method) here.
If it concerns an in-development version, note the branch(es) and commit hash(es) here as well.

- OS: Ubuntu 22.04.4 LTS

### PR DESCRIPTION
**Pull request by Souf149**
### Changes
Python-docker-boefjes' `BoefjeResource` should not get a `module` since the boefje-runner will not execute the boefje in the same context. This would make it that python-docker-boefjes would make OpenKAT crash because OpenKAT will attempt to import the modules the python-docker-boefje requires.

### Issue link
Closes #3582

### Demo
_Please add some proof in the form of screenshots or screen recordings to show (off) new functionality, if there are interesting new features for end-users._

### QA notes
To test the problem, import a non-existing module into a docker-boefje. 
For example inside `boefjes/boefjes/plugins/kat_nmap_tcp/main.py` add the following line at the top of the file
```python
import ihopethispackagedoesnotexist
```
Now when you run a fresh `make kat`, you should notice you are not able to go to the KAT-alogus anymore, and the logs are full of errors.

When doing the same inside this branch this will not be an issue, the issue should only arise when the docker-boefje is being ran.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: boefjes/boefjes/plugins/models.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def __init__(self, path: Path, package: str):
         self.boefje.produces = self.boefje.produces.union(set(_default_mime_types(self.boefje)))
         self.module: Runnable | None = None
 
-        if (path / ENTRYPOINT_BOEFJES).exists():
+        if (path / ENTRYPOINT_BOEFJES).exists() and self.boefje.oci_image is None:
```
Comment (underdarknl) [inline boefjes/boefjes/plugins/models.py:67]: lets check for the image first, thats a cheaper call.

Comment (Souf149) [reply boefjes/boefjes/plugins/models.py]: Good catch, fixed in [6530037](https://github.com/minvws/nl-kat-coordination/pull/3621/commits/65300372c14dfa7306e3114d204cf1980c1408f2)

**Comment by underdarknl** [comment]
looks good to me

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Katalogus loads and also the individual boefje information page. The boefje runs, but fails due to the invalid module, which is expected Python behaviour.

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**